### ISSUE TITLE
JOSM file issue (was Conflict resolution doesn't work)

### ISSUE THREAD

**Issue by Kilkenni** [MarcusWolschon/osmeditor4android#477]
A little background.
As with some other users, I've used f-droid version, and it's not launching (crashing on start) after the last update. I had some changes I didn't upload earlier, so I've saved them into a file, deleted f-droid version and installed the latest Google Play build, then loaded the file I've saved earlier. It's been a day since I mapped this place, so it shouldn't change that much.
What happens next?
1. Vespucci founds a "conflict". Strangely, it declares I have a node (a polygon, a way, anything) that has a local version but does not exist on a server. I know it's not true - while Vespucci shows minimal info about the object, some are easily identified (named buildings, streets, bus stations). And I know for sure those DO exist on a server. I didn't create them, I just modified them. So, what the heck? x1
2. Okay, I press "Use local version". As I understand it, Vespucci should try and push my object to the server anyway. At worst, there will be a duplicate object after that, right?
Wrong. After some "thinking" Vespucci declares THE SAME object doesn't exist on the server. So it pushed nothing. More so, according to conflict resolver, my "local version" now is bumped +1. Every next attempt to push my changes does nothing except bumping local version even higher.
3. If I press "Use server version", the program (if I get it right) deletes my changes for the object and considers the next one. I've checked some of the named objects through JOSM and it seems to be the case - if I say "use server version" my changes are lost.

The example can be seen here:
http://www.openstreetmap.org/#map=19/50.47266/30.47091
As you can see, my changes for the ATM were pushed successfully (I added wheelchair=yes), but my changes for the clinic itself - the building - weren't (I've changed the address, added levels and joined ATM to the wall since it hangs on it - all of it has been ignored when uploading).

**Comment by simonpoole** [comment]
As already pointed out the issues with f-droid are outside of our control.

Which version did you install, the current build from github releases or the one actually on google play? The build number would be helpful.

What was the exact error message you got?


Wrt (2): there will not be a duplicate version but changes to the server may be overridden by using the local version.

Could you attach the file you saved? If I understand you correctly, only one object updated?

**Comment by Kilkenni** [comment]
I installed the one from Google Play. Debug says 0.9.7r1141.
Getting the "exact" error message is tricky as my Vespucci automatically picks up the locale of my smartphone, and I don't know how to force it to use English instead. Can you tell how can I do this? It goes something like: "<name of the object> has a local version <number> but does not exist on the server". When the object has a name, here's where it is show. If not, it uses the "main" tag (in my case it was "building=yes", so I had no idea what it was talking about until it got to named objects I could identify.
If I get it right, when OSM version is changed after you've downloaded the map part, uploading reveals server version number is bigger than your local.
No, there was a bunch of objects. I gave only one example. Some were updated while others weren't. You can look at the whole changeset to get the idea and scope of the changes. I will attach the file as soon as I extract it from my phone.

Also, another unrelated question. When your device turns the screen off while you upload (for example, on a screen timer), the upload is interrupted in the middle of it. If you turn the screen back on and try uploading again, some objects WILL be duplicated as a result. 
Is this the known behavior? I prevent it by keeping the phone from going to sleep when uploading.

**Comment by Kilkenni** [comment]
Here, compressed in a ZIP.
[tatarka.zip](https://github.com/MarcusWolschon/osmeditor4android/files/465608/tatarka.zip)

**Comment by simonpoole** [comment]
@Kilkenni the upload happens in a separate thread (to be exact in an AsyncTask) and should complete independently of what happens to the Main activity. Further in 0.9.7 and earlier, objects are uploaded one by one sequentially, it is not possible that more than one object could be affected by any network interruption or similar. 

Note: if you loaded the data from a file, that file itself is not automatically updated by subsequent uploads or edits (except if you re-save to it) and reloading the -same- file after you have uploaded or edited will naturally produce a gazillion conflicts if you try to upload.

**Comment by simonpoole** [comment]
WRT version: please use the version from github releases, it is signed with the same key as the google play version and has numerous bug fixes including some conflict resolution related ones (or even better use 0.9.8, however the internal format for saving data has changed so you should not have any edits pending for upload when you upgrade).

**Comment by Kilkenni** [comment]
@simonpoole Thanks. So at worst only those objects being uploaded when the upload interrupted coudl potentially be duplicated. Good to know.

**Comment by simonpoole** [comment]
OK, the issue was actually something quite different, there was a bug in reading JOSM style OSM files that caused newly created objects not to be uploaded. Fix see https://github.com/MarcusWolschon/osmeditor4android/commit/532fffdbcd806b4be561eb67f41498da75c7adc8 I can upload your file or you can do it with JOSM or a fixed vespucci version 0.9.7r1159 or 0.9.8b1194. It will likely take a bit before Marcus gets around to signing the APKs, so I'm not quite sure when they will be available.

**Comment by Kilkenni** [comment]
Thanks.
Meanwhile I've found kind of a workaround. The problem is somewhat reproducible, but I figured that when -cancelling- the upload on conflict, the trying to upload -again-, there are usually no isses. Either the version on the server is shown correctly, or the conflict "disappears".
Don't know why it happens but I suspect quality of the internet connection somehow influences it (I mostly use different wifi points with varying signal strength). Will wait for a stable fix.
